assembl/models/section.py

```diff
@@ -0,0 +1,94 @@
+import enum
+from sqlalchemy import (
+    Column,
+    Enum,
+    Float,
+    Integer,
+    String,
+    ForeignKey,
+)
+from sqlalchemy.orm import relationship, backref
+
+from .auth import CrudPermissions, P_READ, P_ADMIN_DISC
+from . import DiscussionBoundBase
+from .langstrings import LangString
+from ..lib.sqla_types import URLString
+
+
+class SectionTypesEnum(enum.Enum):
+
+    HOMEPAGE = 'HOMEPAGE'
+    DEBATE = 'DEBATE'
+    SYNTHESES = 'SYNTHESES'
+    RESOURCES_CENTER = 'RESOURCES_CENTER'
+    CUSTOM = 'CUSTOM'
+    ADMINISTRATION = 'ADMINISTRATION'
+
+
+section_types = [t.value for t in SectionTypesEnum.__members__.values()]
+
+
+class Section(DiscussionBoundBase):
+
+    """Assembl configurable sections."""
+
+    __tablename__ = "section"
+    type = Column(String(60), nullable=False)
+
+    id = Column(Integer, primary_key=True)
+
+    discussion_id = Column(
+        Integer,
+        ForeignKey(
+            'discussion.id',
+            ondelete='CASCADE',
+            onupdate='CASCADE',
+        ),
+        nullable=False, index=True)
+
+    discussion = relationship(
+        "Discussion",
+        backref=backref(
+            'sections',
+            cascade="all, delete-orphan"),
+    )
+
+    title_id = Column(
+        Integer(), ForeignKey(LangString.id))
+    title = relationship(
+        LangString,
+        lazy="joined", single_parent=True,
+        primaryjoin=title_id == LangString.id,
+        backref=backref("section_from_title", lazy="dynamic"),
+        cascade="all, delete-orphan")
+
+    url = Column(URLString)
+
+    section_type = Column(
+        Enum(*section_types, name='section_types'),
+        nullable=False,
+        default=SectionTypesEnum.CUSTOM.value,
+        server_default=SectionTypesEnum.CUSTOM.value
+    )
```

Comment:
You could have avoided the section_type enum alltogether and just used sqlalchemy polymorphism and the type field, no?

Reply:
I don't understand how the type field could have helped me

Reply:
type field is a string. You could have added types like this:
```
class SectionCustom(Section):
    __mapper_args__ = {
        'polymorphic_identity': 'custom'
    }
                                                                                
    @classproperty
    def name(cls): 
        return 'custom'
```

Reply:
But nevermind, keep this enum. We can maybe use type field later to handle footer actions, user actions if there is a need.

Reply:
Agreed. We have consistently used the polymorphic_type on base objects so that we would always have row representation of the type of object instantiated. It's non-critical for now. Can be done ad-hoc down the road.